spec/controllers/infra_networking_controller_spec.rb

```diff
@@ -1,27 +1,121 @@
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+
+        get :explorer
+        expect(response.body).to include("modelName: 'Switch'")
+        expect(response.body).to include("activeTree: 'infra_networking_tree'")
+        expect(response.body).to include("gtlType: 'list'")
+        expect(response.body).to include("isExplorer: 'true' === 'true' ? true : false")
+        expect(response.body).to include("showUrl: '/infra_networking/x_show/'")
+      end
+    end
+
+    describe "#tree_select" do
+      [
+        ['All Distributed Switches', 'infra_networking_tree'],
+      ].each do |elements, tree|
+        it "renders list of #{elements} for #{tree} root node" do
+          session[:settings] = {}
+          seed_session_trees('infra_networking', tree.to_sym)
+
+          post :tree_select, :params => { :id => 'root', :format => :js }
+          expect(response.status).to eq(200)
+        end
+      end
+
+      it "renders textual summary for an infrastructure switch" do
+        allow(switch).to receive(:tagged_with).with(:cat => user.userid).and_return("my tags")
+        allow(Classification).to receive(:find_assigned_entries).with(switch).and_return([tag1, tag2])
+
+        tree_node_id = "sw-#{switch.id}"
+        seed_session_trees('infra_networking', :infra_networking_tree)
+        post :tree_select, :params => { :id => tree_node_id, :tree => :infra_networking_tree, :format => :js }
+
+        expect(response.status).to eq(200)
+        expect(response.body).to include('Department')
+        expect(response.body).to include(tag1.description)
+        expect(response.body).to include(tag2.description)
       end
 
-      it 'renders the network switch center toolbar' do
-        nodeid = [ems, cluster, switch].map { |item| TreeNode.new(item).key }.join('_')
-        expect(ApplicationHelper::Toolbar::InfraNetworkingCenter).to receive(:definition).and_call_original.at_least(:once)
-        post :tree_select, :params => { :id => nodeid }
+      describe 'switch' do
+        let(:ems) { FactoryGirl.create(:ems_vmware) }
+        let(:cluster) { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
+
+        it 'renders the network switch center toolbar' do
+          nodeid = [ems, cluster, switch].map { |item| TreeNode.new(item).key }.join('_')
+          expect(ApplicationHelper::Toolbar::InfraNetworkingCenter).to receive(:definition).and_call_original.at_least(:once)
+          post :tree_select, :params => { :id => nodeid }
+        end
       end
     end
   end
+
+  describe "#tags_edit" do
+    let(:ds) { FactoryGirl.create(:switch, :name => "testSwitch") }
+    before(:each) do
+      EvmSpecHelper.create_guid_miq_server_zone
+      allow(ds).to receive(:tagged_with).with(:cat => user.userid).and_return("my tags")
+      allow(Classification).to receive(:find_assigned_entries).with(ds).and_return([tag1, tag2])
```

Comment:
I think that instead of mocking on the `Classicifacion` singletion you could actually create the tagging on the `ds` record.